frontend/src/concepts/pipelines/content/pipelineSelector/CustomPipelineVersionSelect.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import {
+  Badge,
+  Bullseye,
+  HelperText,
+  HelperTextItem,
+  Menu,
+  MenuContainer,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  MenuSearch,
+  MenuSearchInput,
+  MenuToggle,
+  SearchInput,
+} from '@patternfly/react-core';
+import useDebounceCallback from '~/utilities/useDebounceCallback';
+import PipelineSelectorTableRow from '~/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow';
+import { Table } from '~/components/table';
+import { PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import { pipelineVersionSelectorColumns } from '~/concepts/pipelines/content/pipelineSelector/columns';
+
+type CustomPipelineVersionSelectProps = {
+  versions: PipelineVersionKF[];
+  selection: string | undefined;
+  onSelect: (version: PipelineVersionKF) => void;
+};
+
+const CustomPipelineVersionSelect: React.FC<CustomPipelineVersionSelectProps> = ({
+  versions,
+  selection,
+  onSelect,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+  const [search, setSearch] = React.useState('');
+  const [filteredVersions, setFilteredVersions] = React.useState<PipelineVersionKF[]>(versions);
+  const [visibleLength, setVisibleLength] = React.useState(10);
+  const placeholder =
+    versions?.length === 0 ? 'No versions available' : 'Select a pipeline version';
+
+  const toggleRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+
+  const doSetSearchDebounced = useDebounceCallback(setSearch);
+
+  React.useEffect(() => {
+    if (search) {
+      setFilteredVersions(
+        versions.filter((option) => option.name.toLowerCase().includes(search.toLowerCase())),
+      );
+    } else {
+      setFilteredVersions(versions);
+    }
+    setVisibleLength(10);
+  }, [search, versions]);
+
+  const menu = (
+    <Menu data-id="pipeline-version-selector-menu" ref={menuRef} isScrollable>
+      <MenuContent>
+        <MenuSearch>
+          <MenuSearchInput>
+            <SearchInput
+              value={search}
+              aria-label="Filter pipeline versions"
+              onChange={(_event, value) => doSetSearchDebounced(value)}
+            />
+          </MenuSearchInput>
+          <HelperText>
+            <HelperTextItem variant="indeterminate">{`Type a name to search your ${versions.length} versions.`}</HelperTextItem>
+          </HelperText>
+        </MenuSearch>
+        <MenuList>
+          <div role="menuitem">
+            <Table
+              data-id="pipeline-selector-table-list"
+              emptyTableView={<Bullseye>No results match the filter.</Bullseye>}
```

Comment:
Try to reuse the `EmptyTableView` component like what we did in the PipelineSelector and PipelineVersionSelector
```
emptyTableView={
  <EmptyTableView
    hasIcon={false}
    onClearFilters={onClear}
     variant={EmptyStateVariant.xs}
  />
}
```